### ISSUE TITLE
Proposal: expose ConfigValidator or something like as a public API.

### ISSUE THREAD

**Issue by mysticatea** [eslint/eslint#10289]
We have exposed `RuleTester` then we can use it to test plugin rules and custom rules. However, we haven't exposed any tester for shareable configs and the `configs` property of plugins. I think that we need the official way to test user configs.

I have the test of my configs which use private APIs: https://github.com/mysticatea/eslint-plugin/blob/26907926ddbf06e17e3fb18529e808c7f38872d6/tests/lib/configs/_rules.js
In that test, I needed the following features:

- validate config (includes rule options) schema.
- validate that the config doesn't include non-existence rules.
- validate that the config doesn't include removed rules.
- validate that the config doesn't include deprecated rules.
- load a plugin onto a given name in order to test the plugin before publish.
- find lacking rules for a given plugin.

I'm imaging the following API:

```ts
export interface ConfigTester {
    validate(config: Object, options?: ConfigTester.ValidateOptions): void
    definePlugin(pluginId: string, definition: Object): void
    getRules(): Map<string, Object>
}

export namespace ConfigTester {
    export interface ValidateOptions {
        configName?: string
        reportRemovedRules?: boolean // report rules `eslint/conf/replacements.json` has.
        reportDeprecatedRules?: boolean // report rules `meta.deprecated` is true.
    }
}
```
